auth/auth_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+import (
+	"bytes"
+	"context"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/idtoken"
+	"google.golang.org/api/option"
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestAuthSnippets(t *testing.T) {
+	ctx := context.Background()
+	tc := testutil.SystemTest(t)
+	audience := "iap.googleapis.com"
+
+	want := "Listed all storage buckets."
+
+	buf := &bytes.Buffer{}
+
+	if err := authenticateExplicitWithAdc(buf); err != nil {
+		t.Errorf("authenticateExplicitWithAdc got err: %v", err)
```

Comment:
Use `t.Fatalf` here because subsequent operation(s) depend on this one succeeding?